### PR TITLE
[philips-hue] Auto detect feature

### LINKED ISSUE #952: Philips Hue auto-detect feature
Instead of adding device by model,
is there any way to determine features with device raw data?

According to doc https://github.com/peter-murray/node-hue-api/blob/HEAD/docs/light.md
we might be able to be more generic.

### PR DESCRIPTION
**Pull request by VonOx**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [X] If your changes affects code, did your write the tests?
- [X] Are tests passing? (`npm test` on both front/server)
- [X] Is the linter passing? (`npm run eslint` on both front/server)
- [X] Did you run prettier? (`npm run prettier` on both front/server)

### Description of change

Resolve #952 

Switch is now based on type returned bur hue api.

### CODE REVIEW

File: server/services/philips-hue/lib/light/light.getLights.js [hunk 1]
```diff
@@ -29,52 +29,20 @@ async function getLights() {
...
+        case 'Dimmable light':
           lightsToReturn.push(getPhilipsHueWhiteLight(philipsHueLight, serialNumber, this.serviceId));
           break;
-        case 'LTW012': // Hue White Ambiance E12
-        case 'LTW010': // Hue White & Ambiance Bulb
-        case 'LTW001': // Hue A19 White & Ambiance Bulb
-        case 'LTW013': // Hue White Ambiance GU10
-        case 'LTG002': // Hue White Ambiance GU10 w/ BT
-        case 'LTA001': // Hue White E27 with Bluetooth
+        case 'Color temperature light':
```
Comment (Pierre-Gilles) [inline server/services/philips-hue/lib/light/light.getLights.js:42]: color temperature = getPhilipsHueWhiteTemperatureLight? 
Is it normal ?

Comment (VonOx) [reply server/services/philips-hue/lib/light/light.getLights.js:42]: Yes , devices like LTG002 have this type ( do not ask to me why :/ )

Comment (VonOx) [reply server/services/philips-hue/lib/light/light.getLights.js:42]: image

Comment (VonOx) [reply server/services/philips-hue/lib/light/light.getLights.js:42]: I think I know why , if you change temperature of white bulb , this color change ( yellowish / whitish )

**Review by atrovato — APPROVED** [review]
I'm loving it.

File: server/services/philips-hue/lib/light/light.getLights.js [hunk 2]
```diff
@@ -29,52 +29,20 @@ async function getLights() {
...
-      switch (philipsHueLight.modelid) {
-        case 'LCT007': // hue white & color 2nd generation
-        case 'LCT015': // hue white & color 4th generation
-        case 'LST002': // hue lightstrip indoor 2nd generation
-        case 'LCT024': // Hue play 1
-        case '440400982841': // Hue White and Color Ambiance Play Light Bar
-        case 'LCT010': // Hue A19 White & Color w/ Richer Colors
-        case 'LCA001': // Hue color lamp
-        case 'GL-C-008': // Non-hue LED Strip
-        case 'CLA60 TW OSRAM': // CLA60 TW OSRAM
```
Comment (Pierre-Gilles) [inline server/services/philips-hue/lib/light/light.getLights.js:41]: According to the `lights.json` file, the "CLA60 TW OSRAM" is a `"type": "Color temperature light"`, meaning that with this PR, it'll get a `getPhilipsHueWhiteTemperatureLight` and before it had a `getPhilipsHueColorTemperatureLight`. Was it a mistake before or is it a problem here? (just checking one by one :) )

**Review by Pierre-Gilles — COMMENTED** [review]
Just a small comment, otherwise I'm good with this! :)

File: server/services/philips-hue/lib/light/light.getLights.js (hunk 2 above)
Comment (Pierre-Gilles) [reply server/services/philips-hue/lib/light/light.getLights.js:41]: Ok seems it was a mistake before: https://www.amazon.co.uk/Osram-Lightify-lamp-Tuneable-white/dp/B014OI12V4

Nice catch!

**Review by Pierre-Gilles — APPROVED** [review]

**Pull request merged**